src/main/java/org/sonar/plugins/stash/client/StashClient.java

```diff
@@ -178,6 +185,25 @@ public StashPullRequest getPullRequest(String project, String repository, String
     return StashCollector.extractPullRequest(project, repository, pullRequestId, response);
   }
   
+  public StashPullRequest getPullRequestByBranchName(String project, String repository, String branchName)
+      throws StashClientException {
+    String request = MessageFormat.format(REPO_API + "pull-requests?at=refs/heads/{3}&direction=OUTGOING", baseUrl, project, repository, branchName);
+    
+    JSONObject response = get(request, MessageFormat.format(PULL_REQUEST_GET_FOR_BRANCH_ERROR_MESSAGE, repository, branchName));
+    
+    Long size = (Long) response.get("size");
+    if(size==0) {
+      throw new GitBranchNotFoundOrNotUniqueException("Found no PR for branch "+branchName + " in repo "+repository);
+    } else if(size==1) {

```

Comment:
What if you run a git-flow and make a hotfix commit with PR towards both develop and master (if they happen to be in sync). That case would likely trigger size == 2 and still be valid, right?